jenkins_job_wrecker/modules/triggers.py

```diff
@@ -271,26 +282,42 @@ def ghprbtrigger(top, parent):
         elif child.tag == 'autoCloseFailedPullRequests':
             ghpr['auto-close-on-fail'] = get_bool(child.text)
         elif child.tag == 'blackListCommitAuthor':
-            ghpr['black-list-commit-author'] = child.text.strip().split(' ')
+            if child.text:
+                ghpr['black-list-commit-author'] = child.text.strip().split(' ')
+            else:
+                ghpr['black-list-commit-author'] = []
         elif child.tag == 'blackListLabels':
-            ghpr['black-list-labels'] = child.text.strip().split('\n')
+            if child.text:
+                ghpr['black-list-labels'] = child.text.strip().split('\n')
+            else:
+                ghpr['black-list-labels'] = []
         elif child.tag == 'blackListTargetBranches':
-            ghpr['black-list-target-branches'] = [item[0].text for item in child]
+            black_list_target_branches = [item[0].text for item in child if item[0].text is not None]
+            ghpr['black-list-target-branches'] = black_list_target_branches
         elif child.tag == 'displayBuildErrorsOnDownstreamBuilds':
             ghpr['display-build-errors-on-downstream-builds'] = get_bool(child.text)
         elif child.tag == 'excludedRegions':
-            ghpr['excluded-regions'] = child.text.strip().split('\n')
+            if child.text:
+                ghpr['excluded-regions'] = child.text.strip().split('\n')
+            else:
+                ghpr['excluded-regions'] = []
         elif child.tag == 'includedRegions':
-            ghpr['included-regions'] = child.text.strip().split('\n')
+            if child.text:
+                ghpr['included-regions'] = child.text.strip().split('\n')
+            else:
+                ghpr['included-regions'] = []
         elif child.tag == 'skipBuildPhrase':
             ghpr['skip-build-phrase'] = child.text
         elif child.tag == 'whiteListLabels':
-            ghpr['white-list-labels'] = child.text.strip().split('\n')
+            if child.text:
+                ghpr['white-list-labels'] = child.text.strip().split('\n')
+            else:
+                ghpr['white-list-labels'] = []
         elif child.tag == 'whiteListTargetBranches':
-            ghpr['white-list-target-branches'] = []
-            for branch in child:
-                if branch[0].text is not None:
-                    ghpr['white-list-target-branches'].append(branch[0].text.strip())
+            if child.text:
```

Comment:
Is this the right condition? XML text is not the same as having child elements.

Reply:
Thanks for noticing it. It's not necessary at all.